### PR TITLE
Fix code scanning alert - Prefer COPY over ADD

### LINKED ISSUE #4354: Fix code scanning alert - Prefer COPY over ADD for copying local resources
Prefer COPY over ADD for copying local resources in  addons/dynamic-mount/ossfs/docker/Dockerfile:21

In Dockerfiles, it is recommended to use the COPY instruction over the ADD instruction for copying local resources. The COPY instruction is more straightforward and less prone to errors, making it a better choice for copying local files and directories into a Docker image.

## why
Using the ADD instruction instead of COPY for local resources in Dockerfiles can lead to several issues, including unexpected behavior, increased complexity, and security risks. The ADD instruction has additional features that can introduce unintended side effects, such as automatically extracting compressed files and fetching remote resources. This can make the behavior of the instruction less predictable and harder to understand or even lead to security issues, if, for example, due to a typo in the source path, the ADD instruction could fetch a remote resource instead of copying a local file.

If you only need to copy local files or directories into your Docker image, it is recommended to use the COPY instruction instead. Only use the ADD instruction when you need its additional features, such as fetching remote resources or extracting compressed files. See also the rule {rule:docker:S7026} for more information on using the ADD instruction to fetch remote resources.

### PR DESCRIPTION
**Pull request by omerap12**
### Ⅰ. Describe what this PR does
Fix code scanning alert - Replace `ADD` in Dockerfiles to `COPY` 

### Ⅱ. Does this pull request fix one issue?

fixes #4354 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: addons/nfs/docker/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN apt update && \
     apt clean autoclean && \
     apt autoremove --yes && \
     rm -rf /var/lib/{apt,dpkg,cache,log}/
-ADD ./fluid_config_init.py /
-ADD ./entrypoint.sh /usr/local/bin
+COPY ./fluid_config_init.py /
+COPY ./entrypoint.sh /usr/local/bin
 COPY --from=BUILD /src/fuse/fuse-nfs /bin/fuse-nfs
 CMD ["/usr/local/bin/entrypoint.sh"]
```
Comment (cheyang) [inline addons/nfs/docker/Dockerfile:23]: Please add return of line.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: addons/nfs/docker/Dockerfile (hunk 1 above)
Comment (omerap12) [reply addons/nfs/docker/Dockerfile:23]: Sure :)

**Comment by cheyang** [comment]
@omerap12 Please don't change [vendor/github.com/nxadm/tail/Dockerfile](https://github.com/fluid-cloudnative/fluid/pull/4378#diff-c20a3d650562db23139c78673ffd21a00b1f9c303545612964afb264d8bf9586)

**Comment by omerap12** [comment]
> @omerap12 Please don't change [vendor/github.com/nxadm/tail/Dockerfile](https://github.com/fluid-cloudnative/fluid/pull/4378#diff-c20a3d650562db23139c78673ffd21a00b1f9c303545612964afb264d8bf9586)

Yeap, sorry :(

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**